sound/soc/sof/topology.c

```diff
@@ -952,10 +934,35 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,
 		/* next array */
 		array = (struct snd_soc_tplg_vendor_array *)((u8 *)array
 			+ asize);
+
+		/* move to next target struct */
+		if (found == count) {
+			offset += object_size;
+			total += found;
```

Comment:
a nitpick: we do realise, that `total` will now only be updated every `count / found` iterations, right? So, the `total < count * sets` loop condition becomes less useful. It still should work as long as the data is consistent, but it's good to be aware of this limitation.

Reply:
@lyakh, so the loop should exit when all sets (with count tokens) are found or we run out of parse area. I had to change the conditions this way as the previous one didn't work with spread around "sets", only consecutive. So this should now handle for example target,non-target, target, non-target "set sequence" which I have in mux/demux, and bail out after all sets have been found

Reply:
@juimonen ok, as long as that's the desired and well understood behaviour - it's ok. It's just not very obvious from the loop construct, that a termination criterion wasn't updated upon each iteration.

Reply:
@lyakh so are you good with the PR now?

Reply:
@ranj063 sorry, missed your question. Basically looks fine. I added some suggestions to use `size_t` and maybe a test for `found` overflow could be added.

Reply:
Folks, @juimonen @lyakh this is really hard to follow. If the discussion is closed, please mark individual conversations as closed. I'll close this now, please reopen if this bit is still valid.